clang/test/CodeGenSYCL/stall_enable_device.cpp

```diff
@@ -0,0 +1,56 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+
+// Tests for IR of Intel FPGA [[intel::use_stall_enable_clusters]] function attribute on Device.
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+[[intel::use_stall_enable_clusters]] void test() {}
+
+struct FuncObj {
+  [[intel::use_stall_enable_clusters]] void operator()() const {}
+};
+
+void test1() {
+  auto lambda = []() [[intel::use_stall_enable_clusters]]{};
+  lambda();
+}
+
+class Foo {
+public:
+  [[intel::use_stall_enable_clusters]] void operator()() const {}
+};
+
+int main() {
+  q.submit([&](handler &h) {
+    // Test attribute is presented on function metadata.
```

Comment:
I think these comments are a bit misleading. In every test here, the attribute is remaining attached to spir_func to which attribute is applied. It is not propagated to kernel metadata i.e. spir_kernel. I think you can just remove the comments as the tests are self-explanatory,